tinygrad/dtype.py

```diff
@@ -19,6 +21,9 @@ def vec(self, sz:int):
     if sz == 1 or self.name == 'void': return self  # void doesn't vectorize, and sz=1 is scalar
     return DType(self.priority, self.itemsize*sz, f"{INVERSE_DTYPES_DICT[self.name]}{sz}", None, sz)
   def scalar(self) -> DType: return DTYPES_DICT[self.name[:-len(str(self.count))]] if self.count > 1 else self
+  def __eq__(self, x):
+    return x and (self.priority, self.itemsize, self.name, self.fmt, self.count) == (x.priority, x.itemsize, x.name, x.fmt, x.count)
+  def __lt__(self, x): return (self.priority, self.itemsize, self.name, self.fmt, self.count) < (x.priority, x.itemsize, x.name, x.fmt, x.count)
```

Comment:
I think this should be a separate PR.

When applying the rewrite rules we end up comparing a `DType` to a `PtrDType`.
This isn't supported with the current implementation because
`dataclass(order=True)` enforces that arguments have the same type. What is the
desired behavior here? Should comparing `PtrDType` to `DType` be the same as
comparing `DType` and `DType`? This is how `__eq__` was implemented.